pkg/loki/config_wrapper_test.go

```diff
@@ -0,0 +1,86 @@
+package loki
+
+import (
+	"flag"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/pkg/util/cfg"
+)
+
+func Test_CommonConfig(t *testing.T) {
+	// defaultYamlConfig := `---
+	// server:
+	// port: 8080
+	// `
+	testContext := func(configFileString string, args []string) (ConfigWrapper, ConfigWrapper) {
+		config := ConfigWrapper{}
+		fs := flag.NewFlagSet(t.Name(), flag.PanicOnError)
+
+		file, err := ioutil.TempFile("", "config.yaml")
+		require.NoError(t, err)
+		_, err = file.WriteString(configFileString)
+		require.NoError(t, err)
+
+		configFileArgs := []string{"-config.file", file.Name()}
+		if args == nil {
+			args = configFileArgs
+		} else {
+			args = append(args, configFileArgs...)
+		}
+		err = cfg.DynamicUnmarshal(&config, args, fs)
+		require.NoError(t, err)
+
+		defaults := ConfigWrapper{}
+		freshFlags := flag.NewFlagSet(t.Name(), flag.PanicOnError)
+		err = cfg.DefaultUnmarshal(&defaults, args, freshFlags)
+		require.NoError(t, err)
+
+		return config, defaults
+	}
+
+	t.Run("common path prefix config", func(t *testing.T) {
```

Comment:
TBH, I find table tests hard to read. I think their great for checking relationships between inputs and outputs that resemble a truth table, but for something like this I think it would make it more difficult to understand why a test fails. 

For example, if the test that's testing command line input is still honored fails, the output will give me a line number for that assertion. When I go to that assertion, it won't be immediately clear why it failed because we'll just be passing `test.args` to the function. In order to determine which input actually caused the failure I would have to copy the name of the failing test and then grep for that in the array of inputs to correlate the failure to the correct input data.

Instead, structuring the test like this, all shared logic is still extracted to that `testContext` function, but now when a test fails I'm taken to a line with the failing input data right there above it (ie. `args := []string{"-foo", "bar"}`). IMO this makes it a lot easier to quickly understand why a test fails.